iceoryx_posh/test/moduletests/test_popo_subscriber_port.cpp

```diff
@@ -301,3 +301,165 @@ TEST_F(SubscriberPort_test, nackWhenNotWaitingForResultsInError)
     EXPECT_FALSE(maybeCaproMessage.has_value());
     EXPECT_TRUE(errorHandlerCalled);
 }
+
+class SubscriberPortMultiProducer_test : public Test
+{
+  protected:
+    SubscriberPortMultiProducer_test()
+    {
+    }
+
+    ~SubscriberPortMultiProducer_test()
+    {
+    }
+
+    void SetUp()
+    {
+    }
+
+    void TearDown()
+    {
+    }
+
+    iox::popo::SubscriberPortData m_subscriberPortDataMultiProducer{
+        iox::capro::ServiceDescription("x", "y", "z"),
+        "myApp",
+        iox::cxx::VariantQueueTypes::SoFi_MultiProducerSingleConsumer};
+    iox::popo::SubscriberPortUser m_sutUserSideMultiProducer{&m_subscriberPortDataMultiProducer};
+    iox::popo::SubscriberPortMultiProducer m_sutRouDiSideMultiProducer{&m_subscriberPortDataMultiProducer};
+};
+
+TEST_F(SubscriberPortMultiProducer_test, initialStateNotSubscribed)
+{
+    EXPECT_THAT(m_sutUserSideMultiProducer.getSubscriptionState(), Eq(iox::SubscribeState::NOT_SUBSCRIBED));
+}
+
+TEST_F(SubscriberPortMultiProducer_test, initialStateReturnsNoCaProMessage)
+{
+    auto maybeCaproMessage = m_sutRouDiSideMultiProducer.getCaProMessage();
+
+    EXPECT_FALSE(maybeCaproMessage.has_value());
+}
+
+TEST_F(SubscriberPortMultiProducer_test, subscribeCallResultsInSubCaProMessage)
+{
+    m_sutUserSideMultiProducer.subscribe();
+
+    auto maybeCaproMessage = m_sutRouDiSideMultiProducer.getCaProMessage();
+
+    EXPECT_TRUE(maybeCaproMessage.has_value());
+    auto caproMessage = maybeCaproMessage.value();
+    EXPECT_THAT(caproMessage.m_type, Eq(iox::capro::CaproMessageType::SUB));
+    EXPECT_THAT(caproMessage.m_serviceDescription, Eq(iox::capro::ServiceDescription("x", "y", "z")));
```

Comment:
Move the service description to a struct and check for `Eq(TestServiceDescription)`? x, y and z are magic strings 🦄

Reply:
:moyai: